pkg/osvscanner/osvscanner.go

```diff
@@ -779,19 +797,50 @@ func DoScan(actions ScannerActions, r reporter.Reporter) (models.VulnerabilityRe
 		))
 	}
 
-	// if vulnerability exists it should return error
-	if len(vulnerabilityResults.Results) > 0 {
-		// If any vulnerabilities are called, then we return VulnerabilitiesFoundErr
-		for _, vf := range vulnerabilityResults.Flatten() {
-			if vf.GroupInfo.IsCalled() {
-				return vulnerabilityResults, VulnerabilitiesFoundErr
+	if len(results.Results) > 0 {
+		// Determine the correct error to return.
+		// TODO: in the next breaking release of osv-scanner, consider
+		// returning a ScanError instead of an error.
+		var vuln bool
+		onlyUncalledVuln := true
+		var licenseViolation bool
+		for _, vf := range results.Flatten() {
+			if vf.Vulnerability.ID != "" {
+				vuln = true
+				if vf.GroupInfo.IsCalled() {
+					onlyUncalledVuln = false
+				}
+			}
+			if len(vf.LicenseViolations) > 0 {
+				licenseViolation = true
 			}
 		}
-		// Otherwise return OnlyUncalledVulnerabilitiesFoundErr
-		return vulnerabilityResults, OnlyUncalledVulnerabilitiesFoundErr
-	}
+		onlyUncalledVuln = onlyUncalledVuln && vuln
 
-	return vulnerabilityResults, nil
+		switch {
+		case !vuln && !onlyUncalledVuln && !licenseViolation:
+			// There is no error.
+			return results, nil
+		case vuln && !onlyUncalledVuln && !licenseViolation:
+			return results, VulnerabilitiesFoundErr
+		case !vuln && onlyUncalledVuln && !licenseViolation:
+			// Impossible state.
+			return results, nil
```

Comment:
Just panic here if this should be impossible

Reply:
done